### PR TITLE
Dev docs: move all IDE specific instructions to one page #7490

### LINKED ISSUE #7490: Dev docs: move all IDE specific instructions to one page
Currently, most IDE specific information are in `docs/ide-usage.md` but some more IDE-specific parts are in other docs e.g. `docs/development.md`
May be we should move all such info to one page and point to it from other places?

### PR DESCRIPTION
**Pull request by bjrodrig**
Fixes #7490

Moved IDE specific instructions to ide-usage.md.

Note: on static-analysis.md, in the Tool Stack section, there are some sections that briefly mention Eclipse and IntelliJ, such as indicating where plugins can be found. I did not move these small sections to ide-usage.md, as they were not detailed instructions.

### CODE REVIEW

File: docs/dependencies.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ Add the library based on the name listed in [Maven Central](http://search.maven.
 To update a library's version, simply change the version number declared in `build.gradle` file.
 
 - If you are using command line, your next build will automatically reflect the change.
-- If you are using Eclipse, right click on the project in the Project Explorer and select `Gradle → Refresh Gradle Project` for the changes to be reflected.
```
Comment (LiHaoTan) [inline docs/dependencies.md]: Managing dependencies is not just IDE-related, the way you are doing this right now is to let the developers know how to update dependencies but treating it as black-box unless they somehow visit this document or already know how dependencies are managed.

File: docs/development.md [hunk 2]
```diff
@@ -62,27 +62,7 @@ If you started the server in the background, run the following command to stop i
 
 If the server is running in the foreground, press `Ctrl + C` to stop it or run the above command in a new console.
 
-### With Eclipse
```
Comment (LiHaoTan) [inline docs/development.md]: By moving only the IDE workflow to the IDE document, this document becomes the command-line only work workflow but it is still partially required when you work in an IDE workflow. You have left this in an inconsistent state.

File: docs/troubleshooting-guide.md [hunk 3]
```diff
@@ -22,55 +22,11 @@ Note that some of the screenshots might be outdated, but the instructions will r
   **SOLUTION**: You can re-run the install command again without any side effect. Make sure to choose to update the `PATH` variable this time. Alternatively, you can use other appropriate methods to update your `PATH` variable to include the `/bin` sub-folder of the SDK folder.<br>
```
Comment (LiHaoTan) [inline docs/troubleshooting-guide.md]: Troubleshooting issues are often edge cases or common cases, collapsing them into the IDE section is unnecessarily causing bloat to the IDE document.

**Comment by bjrodrig** [comment]
Thanks for the feedback! What do you think of changing "ide-usage.md" to "ide-setup.md" and leaving the rest of the documents as is, including development.md as wkurniawan07 suggested?

**Comment by LiHaoTan** [comment]
@bjrodrig You can always force push to the same branch and all the changes you did will reset.  Opening a new PR with a different solution seems to be fine here though.

I would say I do agree the easier solution is keeping `ide-usage.md` and `development.md` separate. However, there is still the problem that when setting up an IDE, we have to visit multiple different documents probably in an unstructured manner and that may be inconvenient.

So the original point of view definitely raises a concern that cannot be resolved easily by just combining documents. If you think about it, some possible solutions are having complementary IDE quickstarts or cross-linking of documentation.

Therefore, although renaming `ide-usage.md` to `ide-setup.md` might make sense it is not enough to resolve this issue, although we can consider that a separate issue altogether.

**Comment by wkurniawan07** [comment]
> when setting up an IDE, we have to visit multiple different documents probably in an unstructured manner and that may be inconvenient

The original intention is for anyone who doesn't want to set up any IDE, in which one can simply go straight from `setting-up.md` to `development.md`. Both `setting-up.md` and `ide-usage.md` are not meant to be useful anymore once the setting up phase is over.

> some possible solutions are having complementary IDE quickstarts or cross-linking of documentation

How are these different from the existing ones?

**Comment by LiHaoTan** [comment]
> The original intention is for anyone who doesn't want to set up any IDE, in which one can simply go straight from `setting-up.md` to `development.md`. Both `setting-up.md` and `ide-usage.md` are not meant to be useful anymore once the setting up phase is over, which again strengthens the point why it is more accurate as `ide-setup.md`.

I'm not disputing the fact that it can be better called `ide-setup.md`, but it would be not of much use by just addressing that unless it is the only thing we need to address. I'm also not disagreeing that making IDEs optional can be a good thing.

Originally, `development.md` contains instructions on how to set up IDE related configs, such as the run configurations. With the Google Cloud SDK migration, the run configurations are automatically set up by the plugin.

It appears I have forgotten about the migration, so it does not seem we have to move anything from `development.md` into `ide-usage.md` anymore. My bad here.

> some possible solutions are having complementary IDE quickstarts or cross-linking of documentation

An IDE quickstart will, for example, provide everything that is IDE-related in a single page, whether it is linked or a concise self-contained page. I am not saying whether it is more useful that way but it seems like something that addresses the current issue at hand.

For instance, the setting up of the project, the static analysis, development, and troubleshooting are all on one page.

With that said, I am personally fine with keeping the status quo, and renaming `ide-usage.md` to `ide-setup.md` because it appears to me that keeping IDE-related tasks across different documents seems like a sensible thing to do.

It's whether we want to take that effort to further improve it, which in this case we are not addressing the issue raised or if we don't want to we should close this issue altogether. Of course, I personally think there is not much point in trying to improve the documentation further especially ever since the migration because the effort to impact ratio seems way too small but this is not something up to me.

We can then decide if we want to close the issue because it is now irrelevant or keep the issue as a low priority issue as it currently is.

Resolving the naming of `ide-usage.md` to `ide-setup.md` should be considered a separate issue altogether.

**Comment by bjrodrig** [comment]
Can I delete this pull request since the issue has been closed?

**Comment by LiHaoTan** [comment]
@bjrodrig yes you can delete your branch.

Closing this issue as the issue has been resolved. Thanks for your effort!

**Pull request closed without merging**